libs/remix-ui/run-tab/src/lib/components/instanceContainerUI.tsx

```diff
@@ -1,22 +1,79 @@
 // eslint-disable-next-line no-use-before-define
 import { CustomTooltip } from '@remix-ui/helper'
-import React from 'react'
-import { FormattedMessage, useIntl } from 'react-intl'
+import React, {useEffect, useRef} from 'react'
+import { FormattedMessage } from 'react-intl'
 import { InstanceContainerProps } from '../types'
 import { UniversalDappUI } from './universalDappUI'
 
 export function InstanceContainerUI(props: InstanceContainerProps) {
   const { instanceList } = props.instances
+  const env = useRef()
+
+  useEffect(() => {
+    const fetchSavedContracts = async () => {
+      env.current = await props.plugin.call('blockchain', 'getProvider')
+      if(env.current && env.current === 'injected') {
```

Comment:
could this be for all providers which aren't a Remix VM?

Reply:
This should be discussed